### ISSUE TITLE
Disabled channels should not be actionable (or invisible) when editing a product or variant

### ISSUE THREAD

**Issue by ebarbeito** [Sylius/Sylius#16479]
**Current Behavior**
In the Sylius admin panel, channels can be disabled within the shop configuration. However, when a channel that is associated with some product(s) or variant(s) is disabled, this "disabled channel" remains visible and configurable. It is still possible to enable or disable it for specific products, edit prices, etc.

**Proposal**
Once a channel is disabled (e.g., using the Sylius admin panel `/administration/channels` or via the REST API), the following changes should be implemented when editing or creating a product:

* **Option 1**: The disabled channel(s) are not displayed within the Details section and Pricing tabs. This means there is no option to interact with the configuration of pricing for a channel that has been globally disabled.

* **Option 2**: The disabled channel(s) are visible but no longer actionable. They appear in the Pricing tabs and Channels list, but cannot be clicked or enabled/disabled for a specific product, as the channel is globally disabled.

What do you think about adopting this more restrictive approach to handle disabled channels in the Product and Variant sections of the administration panel? This change would benefit shop administrators by simplifying management, especially for e-commerce platforms with numerous channels and pricing configurations.

Thanks in advance for your consideration.

**Comment by Chrysweel** [comment]
I think that disabled channels should be used in the products to allow for price entry before they are enabled.

However, a disabled channel on the product page can be quite frustrating, as it might cause colleagues managing the products to mistakenly enter data into it.

Therefore, I suggest to provide an option to hide or show disabled channels.

What do you think?

An example of this solution:
![imagen](https://github.com/Sylius/Sylius/assets/1416870/e8c22dbd-96db-44d6-a9c2-dfa990260ecc)

**Comment by Chrysweel** [comment]
Another option is to add a new option.

Disable a channel or "Delete" a channel.

If it is disabled it can appear on the product page, if it is removed it should not appear almost anywhere.

But I think this is the most expensive to implement.

If sylius would accept the option I've proposed, I could make a PR to submit.

**Comment by ebarbeito** [comment]
@Chrysweel: I agree that allowing edits to disabled channels for the use case you’re discussing makes sense. A less restrictive solution to prevent incorrect usage of disabled channels would indeed be more convenient.

Adding a button to toggle the visibility of pricing for disabled channels sounds like a good idea 👍🏼

* On the product page, the list of channels is displayed.
* For channels not selected for the product, their pricing can be hidden.
* If the toggle button is switched on, it would allow editing of these channels while also hiding them to prevent mistakes.

This approach looks good to me. Thanks

**Comment by Chrysweel** [comment]
@CoderMaggie  we could make a PR with the option that is decided, how do you see it? Is this something that would be accepted by sylius?

**Comment by CoderMaggie** [comment]
In my opinion, adding a "Show disabled channels" toggle in the channel pricing is a good idea. You will have my approval when submitting a PR @Chrysweel

**Comment by ebarbeito** [comment]
Hello. I am starting the change to propose a PR when is ready.

As this represents a new feature, the base branch I use for the change proposal is `1.13`. I also will follow the [contributing guidelines](https://docs.sylius.com/en/latest/book/index.html#contributing) for Code and Documentation.
